File: graph/internal/buildpipe/testdata/multilang_v6_markers/backend/wallet.go
// W-C W11 V6 fixture (Go half) — adds a third language so the
// pipeline's auto language detection and per-language parser
// dispatch both run.

package backend

// Wallet is the Go-side mirror of the Sol contract / TS class.
// No binds_to edges are expected from Go to Sol in V6 — the
// linker (T20) covers Sol-TS only — but having a Go file
// guarantees the pipeline exercises all three parsers in one
// build.
type Wallet struct {
	Owner string
}

// Relay is the Go counterpart to Wallet.relay.
func (w *Wallet) Relay(target, data []byte) bool {
	return len(target) > 0 && len(data) > 0
}

File: internal/graph/buildpipe/testdata/multilang_v6_markers/src/Wallet.ts
// W-C W11 V6 fixture (TS half) — same-name class to exercise the
// cross-language binder (T20). The Sol Wallet contract has a
// matching TS class, so the linker should emit at least one
// binds_to edge.

export class Wallet {
  constructor(public owner: string) {}

  // Method that mirrors the Sol contract's relay surface.
  async relay(target: string, data: string): Promise<boolean> {
    return target.length > 0 && data.length > 0;
  }
}
